Mathlib/Data/Complex/Order.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Mathlib.Data.Complex.Module
+
+/-!
+# The partial order on the complex numbers
+
+This order is defined by `z ≤ w ↔ z.re ≤ w.re ∧ z.im = w.im`.
```

Comment:
Feel free to rewrite this as you see fit, I just want a module doc that explains to the uninitiated why we have this at all.
```suggestion
This order is defined by `z ≤ w ↔ z.re ≤ w.re ∧ z.im = w.im`.
This is a natural order on `ℂ` because, as is well-known, there does not exist an
order on `ℂ` making it into a `LinearOrderedField`. However, the order described
above is the canonical order stemming from the structure of `ℂ` as a ⋆-ring (i.e.,
it becomes a `StarOrderedRing`). Moreover, with this order `ℂ` is a `StrictOrderedCommRing` and the coercion `(↑) : ℝ → ℂ` is an order embedding.
```